src/Oauth2CredentialManagers/ModelStore.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Webfox\Xero\Oauth2CredentialManagers;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Session\Store;
+use League\OAuth2\Client\Token\AccessTokenInterface;
+use Webfox\Xero\Oauth2Provider;
+use Webfox\Xero\OauthCredentialManager;
+use Webfox\Xero\Xero;
+
+class ModelStore extends BaseCredentialManager implements OauthCredentialManager
+{
+    public Model $model;
+
+    public function __construct(protected Store $session, protected Oauth2Provider $oauthProvider)
+    {
+        $this->model = Xero::getModelStorage();
+    }
+
+    public function exists(): bool
+    {
+        return $this->model && $this->model->exists && is_array($this->model->{$this->getModelKey()});
+    }
+
+    public function store(AccessTokenInterface $token, array $tenants = null): void
+    {
+        $this->model->update([
+            $this->getModelKey() => [
+                'token' => $token->getToken(),
+                'refresh_token' => $token->getRefreshToken(),
+                'id_token' => $token->getValues()['id_token'],
+                'expires' => $token->getExpires(),
+                'tenants' => $tenants ?? $this->getTenants(),
+            ],
+        ]);
+    }
+
+    protected function data(string $key = null)
+    {
+        if (! $this->exists()) {
+            throw new \Exception('Xero oauth credentials are missing');
```

Comment:
What do you think about updating these to throw a custom `XeroCredentialsNotFound` exception?
Would be easier for devs to respond to.